plugins/inputs/modbus/type_conversions16.go

```diff
@@ -84,54 +29,54 @@ func determineConverterI16Scale(outType, byteOrder string, scale float64) (field
 	case "native":
 		return func(b []byte) interface{} {
 			in := int16(tohost(b))
-			return int16(float64(in) * scale)
+			return int16(float64(in)*scale + shift)
 		}, nil
 	case "INT64":
 		return func(b []byte) interface{} {
 			in := int16(tohost(b))
-			return int64(float64(in) * scale)
+			return int64(float64(in)*scale + shift)
 		}, nil
 	case "UINT64":
 		return func(b []byte) interface{} {
 			in := int16(tohost(b))
-			return uint64(float64(in) * scale)
+			return uint64(math.Max(0, float64(in)*scale+shift))
```

Comment:
Why do you do that here, but not for other operations? Anyway, this changes the current behavior and you should have a good reason for doing so...

Reply:
If the shift is negative and larger than scale × in, we get a negative number and the uint16 conversion leads to very weird results. This is not an issue for the signed data types.

Reply:
Well my point is, you only do it for `uint64` but the same issue (overflow(/underflow)) occurs for all integer types. So either check all or none.